content/source/docs/enterprise/system-overview/architecture.html.md

```diff
@@ -5,24 +5,201 @@ page_title: "Architecture Summary - System Overview - Terraform Enterprise"
+
+1. The first batch of services starts in parallel.
+2. Once `ptfe_nginx` has started, `rabbitmq` starts.
+3. Once `rabbitmq` has started, `postgres` starts.
+
+### Wait for Dependencies
+
+The application waits for `postgres`, `redis`, and `rabbitmq` to start as above. Once these key services respond successfully to requests, the startup process completes with the following steps.
+
+![tfe-int-wait-for-dependencies](assets/tfe-int-wait-for-dependencies.png)
+
+1. The remaining services that do not rely on PostgreSQL are launched in parallel. Included in this batch are or two short lived containers, `ptfe_migrations` and `ptfe_registry_migrations`, which exist only to apply any necessary updates to the database schema.<br />
+2. The main **web** and **background** services start up:
+<br />2a. Once `ptfe_migrations` has finished, `ptfe_atlas` and `ptfe_sidekiq` are started in parallel.
+<br />2b. Once `ptfe_registry_migrations` has finished, `ptfe_registry_api` and `ptfe_registry_worker` are started in parallel.
+
+Finally, startup is complete and the application is ready to go at your configured hostname.
+
+### Module Ingress
+
+The process of importing Terraform modules into the private module registry starts with either a click in the UI or a `POST` to the API. The request is handled the same way in both cases:
+
+![tfe-int-module-ingress](assets/tfe-int-module-ingress.png)
+
+1. `ptfe_atlas` receives request from `ptfe_nginx`, pushes a job to Redis for `ptfe_sidekiq`, and then responds to the original request immediately.
+1. `ptfe_sidekiq` asynchronously pulls the job and passes the details to `ptfe_slug_ingress`.
+1. `ptfe_slug_ingress` downloads your Git repository and `POST`s it to `ptfe_archivist`.
+1. `ptfe_archivist` persists the repository in blob storage and `POST`s the Object ID to `ptfe_atlas`.
+
+
+At this point, each tagged version of your repository has been persisted in the application. The process to make the modules available in your private module registry begins next:
+
+![tfe-int-module-availability](assets/tfe-int-module-availability.png)
+
+1. `ptfe_atlas` sends the Object ID from `ptfe_archivist` to the `ptfe_registry_api`.
+1. `ptfe_registry_api` pulls the module from `ptfe_archivist`, saves the metadata in PostgreSQL, and passes the details to the `ptfe_registry_worker`.
+1. `ptfe_registry_worker` extracts the individual modules from your repository, updates the Private Registry tables in PostgreSQL, and `POST`s the results to `ptfe_atlas`.
+
+Finally, `ptfe_atlas` receives the callback and sets the module’s state to *ingressed* in PostgreSQL. The update is now live in the UI.
+
+### Webhook Receipt
+
+Communications between Terraform Enterprise and a connected VCS server happen via HTTP webhooks.  The latest state of the connected Git repository is downloaded using the process below, and then followup Terraform commands are applied by the state machine.
```

Comment:
Just changing to active voice because it's easier to read - but please reject if this changes the meaning too much!

```suggestion
Communications between Terraform Enterprise and a connected VCS server happen via HTTP webhooks. Terraform downloads the latest state of the connected Git repository using the process below, and then the state machine applies the followup Terraform commands.
```